Assets/GLTF/Scripts/GLTFLoader.cs

```diff
@@ -252,21 +353,110 @@ protected virtual GameObject CreateMeshPrimitive(MeshPrimitive primitive)
-			);
+				if (HasBlendShapes(primitive))
+					SetupBlendShapes(primitive);
+
+				if (HasBones(skin))
+					SetupBones(skin, primitive, skinnedMeshRenderer, primitiveObj);
+
+				skinnedMeshRenderer.sharedMesh = primitive.Contents;
+			}
+			else
+			{
+				var meshFilter = primitiveObj.AddComponent<MeshFilter>();
+				var meshRenderer = primitiveObj.AddComponent<MeshRenderer>();
+				meshFilter.sharedMesh = primitive.Contents;
+				meshRenderer.material = material;
+			}
+		}
 
-			var meshRenderer = primitiveObj.AddComponent<MeshRenderer>();
-			meshRenderer.material = material;
+		bool NeedsSkinnedMeshRenderer(MeshPrimitive primitive, Skin skin)
+		{
+			return HasBones(skin) || HasBlendShapes(primitive);
+		}
+
+		bool HasBones(Skin skin)
+		{
+			return skin != null;
+		}
+
+		void SetupBones(Skin skin, MeshPrimitive primitive, SkinnedMeshRenderer renderer, GameObject primitiveObj)
+		{
+			var boneCount = skin.Joints.Count;
+			Transform[] bones = new Transform[boneCount];
+			Matrix4x4[] bindPoses = skin.InverseBindMatrices.Value.AsMatrix4x4Array();
+
+			Matrix4x4 rightToLeftHanded = new Matrix4x4();
+			rightToLeftHanded.SetRow(0, new Vector4(1, 0, 0, 0));
+			rightToLeftHanded.SetRow(1, new Vector4(0, 1, 0, 0));
+			rightToLeftHanded.SetRow(2, new Vector4(0, 0, -1, 0));
+			rightToLeftHanded.SetRow(3, new Vector4(0, 0, 0, 1));
+
+			for (int i = 0; i < boneCount; i++)
+			{
+				bones[i] = _nodeMap[skin.Joints[i].Id].transform;
+				bindPoses[i] = rightToLeftHanded.inverse * bindPoses[i] * rightToLeftHanded;
+			}
+
+			renderer.rootBone = _nodeMap[skin.Skeleton.Id].transform;
+			primitive.Contents.bindposes = bindPoses;
+			renderer.bones = bones;
+		}
+
+		bool HasBlendShapes(MeshPrimitive primitive)
+		{
+			return primitive.Targets != null;
+		}
+
+		void SetupBlendShapes(MeshPrimitive primitive)
+		{
+			var mesh = primitive.Contents;
+			for (int blendShapeIndex = 0; blendShapeIndex < primitive.Targets.Count; blendShapeIndex++)
+			{
+				var fieldToAccessor = primitive.Targets[blendShapeIndex];
+				var verts = fieldToAccessor["POSITION"].Value.AsVertexArray();
+				var normals = fieldToAccessor["NORMAL"].Value.AsNormalArray();
+				var tangents = fieldToAccessor["TANGENT"].Value.AsVector3Array();
+				mesh.AddBlendShapeFrame(blendShapeIndex.ToString(), 1.0f, verts, normals, tangents);
+			}
+		}
+
+		public BoneWeight[] CreateBoneWeightArray(Vector4[] joints, Vector4[] weights, int vertCount)
+		{
+			var boneWeights = new BoneWeight[vertCount];
+			for(int i = 0; i < vertCount; i++)
+			{
+				boneWeights[i].boneIndex0 = (int)joints[i].x;
+				boneWeights[i].boneIndex1 = (int)joints[i].y;
+				boneWeights[i].boneIndex2 = (int)joints[i].z;
+				boneWeights[i].boneIndex3 = (int)joints[i].w;
+
+				//If weights don't sum to 1, we need to adjust them up until they do.
+				var weightSum = (weights[i].x + weights[i].y + weights[i].z + weights[i].w);
+				if (weightSum < 0.98f)
+					weights[i] /= weightSum;
+
```

Comment:
Can we extract a method called "normalize" for this?

Reply:
No, because, this isn't normalizing. There already is a normalize function for the Vector4, but our goal is to make the sum of its values equal 1, not the length of the constituted vector.

That said, this does seem like a  good culprit for a helper function.